### ISSUE TITLE
Ttd value not given downstream

### ISSUE THREAD

**Issue by calohmn** [eclipse/hono#1062]
Consider the following scenario:
- Deploy Hono with the `swarm_deploy.sh` script
- Start the `HonoExampleApplication`
- Upload Telemetry Data using the HTTP Adapter with the `hono-ttd` param - _without having registered the device yet!_
- Register the device
- Upload Telemetry Data again - with the `hono-ttd` param

=> The `HonoExampleApplication` doesn't receive the Ttd value.

**Comment by calohmn** [comment]
HTTP adapter log from the first request (device not registered yet):
````
16:35:22.462 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - sent request [target address: registration/DEFAULT_TENANT, subject: assert, correlation ID: reg-client-99d33a81-694e-402a-be4d-82ef744e5b3f, device ID: 4711] to service
16:35:22.487 [vert.x-eventloop-thread-0] DEBUG o.e.h.client.impl.AbstractHonoClient - sender open [target: tenant, sendQueueFull: false]
16:35:22.488 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - request-response client for peer [hono-registry-0.0.0.0:5671] created
16:35:22.489 [vert.x-eventloop-thread-0] DEBUG o.e.h.client.impl.TenantClientImpl - successfully created tenant client
16:35:22.489 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - successfully created new client [target: tenant]
16:35:22.504 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - sent request [target address: tenant, subject: get, correlation ID: tenant-client-efeb6a3e-0f90-42f9-887d-ab57152a26d6] to service
16:35:22.527 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - received response [reply-to: registration/DEFAULT_TENANT/ff017d9a-1b47-4c0e-92db-34244a65b31e, subject: null, correlation ID: reg-client-99d33a81-694e-402a-be4d-82ef744e5b3f, status: 404]
16:35:22.530 [vert.x-eventloop-thread-0] DEBUG o.e.h.a.h.v.VertxBasedHttpProtocolAdapter - cannot process [TELEMETRY] message from device [tenantId: DEFAULT_TENANT, deviceId: 4711]
org.eclipse.hono.client.ClientErrorException: device unknown or disabled
	at org.eclipse.hono.client.impl.RegistrationClientImpl.lambda$assertRegistration$7(RegistrationClientImpl.java:363)
	at io.vertx.core.Future.lambda$map$2(Future.java:301)
	at io.vertx.core.impl.FutureImpl.tryComplete(FutureImpl.java:125)
...
	at org.eclipse.hono.client.impl.AbstractRequestResponseClient.handleResponse(AbstractRequestResponseClient.java:486)
	at org.eclipse.hono.client.impl.AbstractHonoClient.lambda$createReceiver$8(AbstractHonoClient.java:394)
	at io.vertx.proton.impl.ProtonReceiverImpl.onDelivery(ProtonReceiverImpl.java:206)
	at io.vertx.proton.impl.ProtonTransport.handleSocketBuffer(ProtonTransport.java:163)
	at io.vertx.core.net.impl.NetSocketImpl$DataMessageHandler.handle(NetSocketImpl.java:392)
...
16:35:22.533 [vert.x-eventloop-thread-0] DEBUG o.e.h.s.http.DefaultFailureHandler - handling failed route for request [method: POST, URI: http://192.168.99.100:8080/telemetry?hono-ttd=30]
16:35:22.615 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - received response [reply-to: tenant/a714ce71-1f00-43fc-9f81-54613e7133f8, subject: null, correlation ID: tenant-client-efeb6a3e-0f90-42f9-887d-ab57152a26d6, status: 200]
16:35:22.629 [vert.x-eventloop-thread-0] DEBUG o.e.h.a.h.v.VertxBasedHttpProtocolAdapter - protocol adapter [hono-http] is enabled for tenant [DEFAULT_TENANT]
16:35:22.634 [vert.x-eventloop-thread-0] INFO  o.e.hono.client.CommandConsumer - creating new command consumer [tenant-id: DEFAULT_TENANT, device-id: 4711], address: control/DEFAULT_TENANT/4711
16:35:22.694 [vert.x-eventloop-thread-0] DEBUG o.e.h.client.impl.AbstractHonoClient - receiver open [source: control/DEFAULT_TENANT/4711]
16:35:22.694 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.CommandConsumer - successfully created command consumer [control/DEFAULT_TENANT/4711]
````
Note: the command consumer is created here nonetheless - and it is not closed.

----
HTTP adapter log from the second request:
````
16:35:55.706 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - reusing existing client [target: credentials/DEFAULT_TENANT]
16:35:55.712 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - sent request [target address: credentials/DEFAULT_TENANT, subject: get, correlation ID: cred-client-a9ee2558-f3fc-48d9-bc59-9ff0a3d0c291] to service
16:35:55.718 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - received response [reply-to: credentials/DEFAULT_TENANT/88e5a751-d070-4bd8-b988-4b42752d703a, subject: null, correlation ID: cred-client-a9ee2558-f3fc-48d9-bc59-9ff0a3d0c291, status: 200]
16:35:55.724 [vert.x-worker-thread-1] DEBUG o.e.h.s.a.d.UsernamePasswordAuthProvider - validating password hash on vert.x worker thread [vert.x-worker-thread-1]
16:35:55.728 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - reusing existing message sender [target: telemetry/DEFAULT_TENANT, credit: 250]
16:35:55.729 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - reusing existing client [target: registration/DEFAULT_TENANT]
16:35:55.735 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - sent request [target address: registration/DEFAULT_TENANT, subject: assert, correlation ID: reg-client-a5257f08-e8f3-4feb-bf58-c2b61df8a774, device ID: 4711] to service
16:35:55.739 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - reusing existing client [target: tenant]
16:35:55.740 [vert.x-eventloop-thread-0] DEBUG o.e.h.a.h.v.VertxBasedHttpProtocolAdapter - protocol adapter [hono-http] is enabled for tenant [DEFAULT_TENANT]
16:35:55.754 [vert.x-eventloop-thread-0] INFO  o.e.hono.service.http.HttpUtils - XXX ttd=30

16:35:55.755 [vert.x-eventloop-thread-0] DEBUG o.e.hono.client.impl.HonoClientImpl - cannot create concurrent command consumer [tenant: DEFAULT_TENANT, device-id: 4711]

16:35:55.794 [vert.x-eventloop-thread-0] DEBUG o.e.h.c.i.AbstractRequestResponseClient - received response [reply-to: registration/DEFAULT_TENANT/ff017d9a-1b47-4c0e-92db-34244a65b31e, subject: null, correlation ID: reg-client-a5257f08-e8f3-4feb-bf58-c2b61df8a774, status: 200]
16:35:55.796 [vert.x-eventloop-thread-0] INFO  o.e.h.a.h.v.VertxBasedHttpProtocolAdapter - XXX addProperties: timeUntilDisconnect=null
````

**Comment by sophokles73** [comment]
@calohmn thanks for spotting, this looks like a bug

**Comment by sophokles73** [comment]
fixed in 3589cef6d0eeabab96c2e09f7b0d1bad78eb1cd0
@calohmn can you confirm and close?

**Comment by calohmn** [comment]
Confirmed.
